doc/dev/sharp_bits.rst

```diff
@@ -810,6 +810,11 @@ array(0.2)
 
     The Catalyst ``@qjit`` decorator doesn't yet support this functionality.
 
+Dynamically-shaped arrays
+-------------------------
+
+TODO.
+
```

Comment:
Todo: this depends on the status of 

- Indexing: https://github.com/PennyLaneAI/catalyst/pull/411
- Control flow: https://github.com/PennyLaneAI/catalyst/pull/370

Reply:
(cc @dime10 @grwlf)

Reply:
@dime10 I'm assuming this will not be merged in, so have written this section with these restrictions mentioned.

Reply:
Yup, looks good 👍